packages/tables/resources/js/components/table.js

```diff
@@ -157,5 +167,54 @@ export default function table() {
         resetCollapsedGroups: function () {
             this.collapsedGroups = []
         },
+
+        initializeCheckboxes: function () {
+
+            let checkboxes =
+                this.$root?.getElementsByClassName('fi-ta-record-checkbox') ??
+                []
+            for (let checkbox of checkboxes) {
+                checkbox.removeEventListener('click', this.handleCheckboxClick) // Remove existing listener to avoid duplicates
+                checkbox.addEventListener('click', (event) =>
+                    this.handleCheckboxClick(event, checkbox),
+                )
+            }
+        },
+
+        handleCheckboxClick: function (event, checkbox) {
+            if (!this.lastChecked) {
+                this.lastChecked = checkbox
+                return
+            }
+
+            if (event.shiftKey) {
+                let checkboxes = Array.from(
+                    this.$root?.getElementsByClassName(
+                        'fi-ta-record-checkbox',
+                    ) ?? [],
+                )
+
+                if (!checkboxes.includes(this.lastChecked)) {
+                    this.lastChecked = checkbox
+                    return
+                }
+
+                let start = checkboxes.indexOf(this.lastChecked)
+                let end = checkboxes.indexOf(checkbox)
+
+                let range = [start, end].sort((a, b) => a - b)
+                for (let i = range[0]; i <= range[1]; i++) {
+                    checkboxes[i].checked = checkbox.checked
+                    let value = checkboxes[i].value
+                    if (checkbox.checked) {
+                        this.selectRecords([value])
+                    } else {
+                        this.deselectRecords([value])
+                    }
+                }
```

Comment:
Instead of calling `selectRecords()` and `deselectRecords()` multiple times, is it possible to construct an array of checkboxes to be selected and an array to be deselected, and then call each method just once after the loop has finished?

Reply:
Good catch, fixed in https://github.com/filamentphp/filament/pull/13074/commits/3728b38cf95bee57f8411b5a7ae1ac1ade7a3065